src/Moq/Linq/ConstructorCallVisitor.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Moq.Properties;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Moq.Linq
+{
+	internal class ConstructorCallVisitor : ExpressionVisitor
```

Comment:
This new class & file do not belong here, `Moq.Linq` is for stuff related to `Mock.Of<>`. Please move it either to `Moq` or (perhaps more accurately) in `Moq.Expressions.Visitors`.